### PR TITLE
Reconstruct V1 messages in storage-proto instead of downgrading them to V0

### LINKED ISSUE #14873: storage-proto: V1 messages decode as V0, silently dropping the transaction config
#### Problem

`impl From<generated::Message> for VersionedMessage` in `storage-proto/src/convert.rs` decides the message version by branching on the `versioned` boolean alone and never reads `value.config`:

```rust
if !value.versioned {
    Self::Legacy(LegacyMessage { /* ... */ })
} else {
    Self::V0(v0::Message { /* ... */ })
}
```

`versioned` is true for both V0 and V1, so every V1 message decodes back as `VersionedMessage::V0`: the version is misreported and the inline compute-budget request (`config`) is silently dropped. No error is raised.

The outbound direction in the same file is correct — `impl From<v1::Message> for generated::Message` writes `config: Some(...)`. Both halves were added in the same commit (edd5368eda, "Sdk upgrades for txv1 (#10699)"); only the write direction handles V1.

Presence of `config` is the only signal on the wire separating V0 from V1: `config` is written only for V1 messages, whose lifetime specifier travels in `recent_blockhash` and which carry no address table lookups.

Beyond the missing field, a V1 message reconstructed as V0 no longer corresponds to the bytes that were signed: re-serializing it produces a different message hash, and the attached signature does not verify against it.

Affected: published `solana-storage-proto` 4.2.0 and current `master` (4.4.0-alpha.1). This impl is the only inbound message conversion, reached from `From<generated::Transaction> for VersionedTransaction`. Downstream, `rpcpool/yellowstone-grpc`'s Node client decodes through this conversion, so its `txEncode.encode` reports V1 transactions as `version: 0` with `transactionConfig` absent (verified against `@triton-one/yellowstone-grpc` 6.0.0 with a V1 transaction produced on `solana-test-validator` 4.2.1). yellowstone-grpc fixed the identical bug in their own copy of this conversion by checking `config` before `versioned` (rpcpool/yellowstone-grpc#855, released in `yellowstone-grpc-geyser` 15.1.1).

#### Proposed Solution

In `From<generated::Message> for VersionedMessage`, check `value.config` before the `versioned` branch: when `config` is present, reconstruct `VersionedMessage::V1`, mapping the four config fields and using `recent_blockhash` as the `lifetime_specifier`. Legacy and V0 decoding stay unchanged. Add round-trip tests (`VersionedMessage -> generated::Message -> VersionedMessage`) for legacy, V0, V1, and V1 with an empty config — a V1 message may request nothing at all and must still be distinguishable from V0.

I have a fix with tests ready and will open a PR.

### PR DESCRIPTION
**Pull request by serejke**
#### Problem

`impl From<generated::Message> for VersionedMessage` in `storage-proto/src/convert.rs` decides the message version by branching on the `versioned` boolean alone and never reads `value.config`. `versioned` is true for both V0 and V1, so every V1 message decodes back as `VersionedMessage::V0`: the version is misreported and the inline compute-budget request (`config`) is silently dropped. No error is raised.

The outbound direction in the same file is correct — `impl From<v1::Message> for generated::Message` writes `config: Some(...)`. Both halves were added in the same commit (edd5368eda, #10699); only the write direction handles V1.

Beyond the missing field, a V1 message reconstructed as V0 no longer corresponds to the bytes that were signed: re-serializing it produces a different message hash, and the attached signature does not verify against it.

This affects published `solana-storage-proto` 4.2.0 and current master. Downstream, `rpcpool/yellowstone-grpc`'s Node client decodes through this conversion, so its `txEncode.encode` reports V1 transactions as `version: 0` with `transactionConfig` absent. yellowstone-grpc fixed the identical bug in their own copy of this conversion (rpcpool/yellowstone-grpc#855) by checking `config` before `versioned`.

#### Summary of Changes

* In `From<generated::Message> for VersionedMessage`, return `VersionedMessage::V1` early when `value.config` is present, mapping the four config fields and using `recent_blockhash` as the `lifetime_specifier`. Presence of `config` is the only signal on the wire separating V0 from V1: it is written only for V1 messages, whose lifetime specifier travels in `recent_blockhash` and which carry no address table lookups.
* Move the `address_table_lookups` computation below the V1 early return (V1 messages have none). Legacy and V0 decoding are unchanged.
* Add round-trip tests (`VersionedMessage -> generated::Message -> VersionedMessage`) for legacy, V0, V1, and V1 with an empty config — a V1 message may request nothing at all and must still be distinguishable from V0.

#### Testing

* `cargo test -p solana-storage-proto --features agave-unstable-api` — all 20 tests pass.
* With the fix reverted and the new tests kept, `v1_message_round_trips` and `v1_message_with_empty_config_round_trips` fail with `V0(...)` on the left and `V1(...)` on the right; the legacy and V0 round-trip tests still pass.
* `cargo fmt -p solana-storage-proto -- --check` and `cargo clippy -p solana-storage-proto --features agave-unstable-api --all-targets` are clean.

Fixes #14873

### CODE REVIEW

File: storage-proto/src/convert.rs [hunk 1]
```diff
@@ -2052,3 +2071,98 @@ mod test {
         }
     }
 }
+
+#[cfg(test)]
+mod message_conversion_tests {
```
Comment (apfitzge) [inline storage-proto/src/convert.rs]: why not as part of the existing `test` module?

**Review by apfitzge — COMMENTED** [review]
Very small nit - fix is small and tests show the fix works.

File: storage-proto/src/convert.rs (hunk 1 above)
Comment (serejke) [reply storage-proto/src/convert.rs]: No good reason — moved it into `mod test`, helpers at the top.

File: storage-proto/src/convert.rs [hunk 2]
```diff
@@ -2051,4 +2098,63 @@ mod test {
...
+    /// presence of `config`. Reading it back as V0 would silently drop the compute-budget request
+    /// and misreport the version.
+    #[test]
+    fn v1_message_round_trips() {
+        let message = VersionedMessage::V1(v1::Message {
+            header: header(),
+            config: v1::TransactionConfig::empty()
+                .with_priority_fee(5_000)
+                .with_compute_unit_limit(30_000)
+                .with_loaded_accounts_data_size_limit(200_000),
```
Comment (bw-solana) [inline storage-proto/src/convert.rs]: could probably add even more validation here including `with_heap_size`, but this is clearly an improvement

**Review by bw-solana — DISMISSED** [review]

File: storage-proto/src/convert.rs [hunk 3]
```diff
@@ -414,6 +414,25 @@ impl From<generated::Message> for VersionedMessage {
             .map(Hash::new_from_array)
             .unwrap();
         let instructions = value.instructions.into_iter().map(|ix| ix.into()).collect();
+
+        // `config` is written only for V1 messages, whose lifetime specifier travels in
```
Comment (t-nelson) [inline storage-proto/src/convert.rs:418]: presumably this is only true until v2 messages exist? we left a similar landmine for ourselves ~ln417

File: storage-proto/src/convert.rs (hunk 2 above)
Comment (serejke) [reply storage-proto/src/convert.rs]: Added `with_heap_size` to the round-trip.

File: storage-proto/src/convert.rs (hunk 3 above)
Comment (serejke) [reply storage-proto/src/convert.rs:418]: Yep, same shape as `versioned`. The proto carries no version number, so the reader can only infer, and this holds until a later version also carries a config. The proper fix is an explicit version field on `Message` — master-only, so I kept it out of this one. Happy to file it separately.

Comment (t-nelson) [reply storage-proto/src/convert.rs:418]: > The proto carries no version number

lmao. we hate ourselves

> Happy to file it separately

yes please!

Comment (apfitzge) [reply storage-proto/src/convert.rs:418]: the fact that any of this exists at all is a huge mistake - we should strongly consider just storing the bytes not any of this protobuf translation.

Unless we convert old bigtable we have to maintain the existing, but we should probably just do bytes for all future transactions. It's literally non-sense and even less byte efficient.

Comment (t-nelson) [reply storage-proto/src/convert.rs:418]: brother let me introduce you to the entirety of the rpc interface

Comment (serejke) [reply storage-proto/src/convert.rs:418]: Filed as #14903. Small, additive, and if it lands before the txv1 gate flips, every V1 row in the archive carries its version from day one. I can open the PR right away if you're okay with the shape.

**Review by apfitzge — APPROVED** [review]

File: storage-proto/src/convert.rs (hunk 3 above)
Comment (apfitzge) [reply storage-proto/src/convert.rs:418]: @t-nelson are we gonna block on making this code better? This PR as a standalone unblocks txv1. If we're gonna require the either version or bytes transition then we should make it part of this PR.

LMK before I merge - aiming for backport on whatever solution here.

Comment (t-nelson) [reply storage-proto/src/convert.rs:418]: no
